tests/compile-test.rs

```diff
@@ -114,6 +126,22 @@ fn run_ui_toml() {
     let config = config("ui", path);
     let tests = compiletest::make_tests(&config);
 
+    let tests = tests
+        .into_iter()
+        .map(|test| {
+            libtest::TestDescAndFn {
+                desc: libtest::TestDesc {
+                    name: libtest::TestName::DynTestName(test.desc.name.to_string()),
+                    ignore: test.desc.ignore,
+                    allow_fail: test.desc.allow_fail,
+                    should_panic: libtest::ShouldPanic::No,
+                },
+                // oli obk giving up
+                testfn: unsafe { std::mem::transmute(test.testfn) },
```

Comment:
I would put up an issue about this, there's probably someone who already knows how to write it without `transmute` right? :thinking:

Reply:
I think oli-obk meant opening issues only for broken tests, it's probably not possible to avoid `transumte` until `libtest` gets fixed.

Reply:
Ah, I see. It's just one line though, it doesn't seem that bad right?